### PR TITLE
RavenDB-20070

### PR DESCRIPTION
**Pull request by ppekrol**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-20070

### Type of change

- New feature

### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.

### Testing by Contributor

- Tests have been added that prove the fix is effective or that the feature works

### Testing by RavenDB QA team

- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/DefaultRavenSerializationBinder.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        if (type == null)
+            return;
+
+        if (SafeTypesCache.Contains(type))
+            return;
+
+        if (ForbiddenTypesCache.Contains(type))
+            ThrowForbiddenType(type);
+
+        if (ForbiddenNamespaces.Contains(type.Namespace))
```
Comment (ayende) [inline src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/DefaultRavenSerializationBinder.cs]: This is wrong, no? 
This needs to be `StartsWith`, I think.

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/DefaultRavenSerializationBinder.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using Newtonsoft.Json.Serialization;
+
+namespace Raven.Client.Json.Serialization.NewtonsoftJson.Internal;
+
+internal class DefaultRavenSerializationBinder : DefaultSerializationBinder
```
Comment (ayende) [inline src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/DefaultRavenSerializationBinder.cs]: We can make this public, and then users can directly add to `SafeTypes`, etc if they need to.
Easier than creating their own serialization binders.

File: src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/DefaultRavenSerializationBinder.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/DefaultRavenSerializationBinder.cs]: Went with full namespace, but after re-thinking you are correct, changed to StartsWith

File: src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/DefaultRavenSerializationBinder.cs (hunk 2 above)
Comment (ppekrol) [reply src/Raven.Client/Json/Serialization/NewtonsoftJson/Internal/DefaultRavenSerializationBinder.cs]: Done, can you check if this would work?

**Review by ayende — APPROVED** [review]

**Review by arekpalinski — APPROVED** [review]

**Pull request merged**